source/logrepl/internal/subscription.go

```diff
@@ -184,11 +184,15 @@ func (s *Subscription) handlePrimaryKeepaliveMessage(ctx context.Context, copyDa
 	if err != nil {
 		return fmt.Errorf("failed to parse primary keepalive message: %w", err)
 	}
+
+	atomic.StoreUint64((*uint64)(&s.serverWALEnd), uint64(pkm.ServerWALEnd))
```

Comment:
Seems like we can get away without the `atomic` function, since `serverWALEnd` will only ever be read/written by a single goroutine. Right?

Reply:
@lovromazgon yea, except my test was causing a data race by reading it for comparison.